### PR TITLE
Added support for cluster roll.

### PR DESCRIPTION
**Pull request by sharadkesarwani**
Added support for cluster roll.

# Jira Ticket
https://spotinst.atlassian.net/browse/SPOTAUT-17745

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -252,4 +252,57 @@ variable "shutdown_hours" {
...
+variable "should_roll" {
+  type        = bool
+  default     = false
+  description = "Should the cluster be rolled for configuration updates"
+}
+## roll_config ##
+variable "conditioned_roll" {
+  type        = bool
+  default     = null
+  description = "Default: false. Spot will perform a cluster Roll in accordance with a relevant modification of the cluster’s settings. When set to true , only specific changes in the cluster’s configuration will trigger a cluster roll (such as availability_zones, max_pods_per_node, enable_node_public_ip, os_disk_size_gb, os_disk_type, os_sku, kubernetes_version, vnet_subnet_ids, pod_subnet_ids, labels, taints, tags)."
```
Comment (chandra1-n) [inline variables.tf:267]: default value and description doesn't match, please modify

File: variables.tf [hunk 2]
```diff
@@ -252,4 +252,57 @@ variable "shutdown_hours" {
...
+}
+variable "respect_pdb" {
+  type        = bool
+  default     = null
+  description = "Default: true. During the roll, if the parameter is set to True we honor PDB during the instance replacement."
```
Comment (chandra1-n) [inline variables.tf]: default value and description doesn't match, please modify

File: variables.tf [hunk 3]
```diff
@@ -252,4 +252,57 @@ variable "shutdown_hours" {
...
+}
+variable "vng_ids" {
+  type        = list(string)
+  default     = null
+  description = "List of virtual node group identifiers to be rolled. Each identifier is a string. vngIds can be null, and cannot be used together with nodeNames and nodePoolNames."
+}
+variable "batch_min_healthy_percentage" {
+  type        = number
+  default     = null
+  description = "Default: 50. Indicates the threshold of minimum healthy nodes in single batch. If the amount of healthy nodes in single batch is under the threshold, the roll will fail. If exists, the parameter value will be in range of 1-100. In case of null as value, the default value in the backend will be 50%. Value of param should represent the number in percentage (%) of the batch."
```
Comment (chandra1-n) [inline variables.tf:287]: default value and description doesn't match, please modify

File: examples/basic/main.tf [hunk 4]
```diff
@@ -55,5 +55,7 @@ module "ocean-aks-np" {
   vmsizes_filters_min_disk                 = 1
   shutdown_hours                           = { is_enabled = false,
                                                time_windows = ["Fri:15:30-Sat:13:30", "Sun:15:30-Mon:13:30"] }
+  should_roll                              = true
+  batch_size_percentage                    = 33
```
Comment (chandra1-n) [inline examples/basic/main.tf]: update the value to default

File: examples/basic/main.tf [hunk 5]
```diff
@@ -55,5 +55,7 @@ module "ocean-aks-np" {
   vmsizes_filters_min_disk                 = 1
   shutdown_hours                           = { is_enabled = false,
                                                time_windows = ["Fri:15:30-Sat:13:30", "Sun:15:30-Mon:13:30"] }
+  should_roll                              = true
```
Comment (chandra1-n) [inline examples/basic/main.tf]: keep the value as false

File: variables.tf [hunk 6]
```diff
@@ -252,4 +252,57 @@ variable "shutdown_hours" {
...
+}
+variable "node_pool_names" {
+  type        = list(string)
+  default     = null
+  description = "List of node pools to be rolled. Each node pool name is a string. nodePoolNames can be null, and cannot be used together with nodeNames and vngIds."
+}
+variable "respect_restrict_scale_down" {
+  type        = bool
+  default     = null
+  description = "Default: false. During the roll, if the parameter is set to true we honor Restrict Scale Down label during the nodes replacement."
```
Comment (chandra1-n) [inline variables.tf:302]: default value and description doesn't match, please modify

**Review by chandra1-n — CHANGES_REQUESTED** [review]
please fix the review comments

File: examples/basic/main.tf (hunk 4 above)
Comment (sharadkesarwani) [reply examples/basic/main.tf]: Updated to 25, as per the the value mentioned in API doc.

File: variables.tf (hunk 1 above)
Comment (sharadkesarwani) [reply variables.tf:267]: Updated value from null to default value.

File: variables.tf (hunk 3 above)
Comment (sharadkesarwani) [reply variables.tf:287]: Updated value from null to default value.

File: examples/basic/main.tf (hunk 5 above)
Comment (sharadkesarwani) [reply examples/basic/main.tf]: updated to false

File: variables.tf (hunk 6 above)
Comment (sharadkesarwani) [reply variables.tf:302]: Updated value from null to default value.

File: variables.tf (hunk 2 above)
Comment (sharadkesarwani) [reply variables.tf]: Updated value from null to default value.

**Review by chandra1-n — APPROVED** [review]

**Pull request merged**